stdlib/string.ml

```diff
@@ -100,15 +100,17 @@ let trim s =
     then bts (B.trim (bos s))
   else s
 
-let escaped s =
-  let rec needs_escape i =
-    if i >= length s then false else
-      match unsafe_get s i with
-      | '\"' | '\\' | '\n' | '\t' | '\r' | '\b' -> true
-      | ' ' .. '~' -> needs_escape (i+1)
-      | _ -> true
+let needs_escape s =
+  let rec aux s n i =
+    if i >= n then false else
+      match String.unsafe_get s i with
+      | '\"' | '\\' | '\000'..'\031' | '\127'.. '\255' -> true
+      | _ -> aux s n (i+1)
   in
-  if needs_escape 0 then
+  aux s (String.length s) 0
+
+let escaped s =
+  if needs_escape s then
```

Comment:
On second thought (sorry), the code could be simplified by directly calling from `escaped` the recursive function (with the three arguments).  It probably improves even a bit the generated bytecode.